activerecord/test/cases/associations/inverse_associations_test.rb

```diff
@@ -872,6 +872,16 @@ def test_building_has_many_parent_association_inverses_one_record
       assert_equal 1, interest.human.interests.size
     end
   end
+
+  def test_dup_with_inversing_does_not_create_additional_records_when_owner_already_exists
```

Comment:
Do you know if there are tests for cases where one of the three `record.new_record? && inversing && !owner.new_record?` isn't true? If not we should add those as well.

Reply:
I added more tests that have hopefully more descriptive names to cover cases for when one of `record.new_record? && inversing && !owner.new_record?` is not true. I didn't see existing tests that covered those situations.

The new tests point to a problem though. My proposed solution of checking `inverse_of_existing_owner?` breaks inversing for unsaved objects. This test demonstrates that:
```ruby
  def test_existing_parent_and_new_child_with_inversing_does_not_create_additional_associations
    with_has_many_inversing(Interest) do
      human = Human.create!
      interest = Interest.new(human: human)
      human.update!(name: "name")
      assert_equal human, interest.human
      assert_equal 0, human.interests.size
    end
  end
```

On the current `main` branch, `human.interests.size` should be `1` because of the inversing between the models. So my proposed solution is no good. 

I went around in circles on this point and along the way discovered that `inverse_of` results in autosaving of associated objects. It's unclear to me if it's supposed to, though I looked through the documentation for clarity on that point. 

I pushed up the new commits to show the process I went through in case it's beneficial to solving the original issue, but I imagine this pull request can be closed.